File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int n = 0;
        for(int i = 0; i<nums.length; i++){
            if(nums[i] != 0){
                nums[n] = nums[i];
                n += 1;
            }
        }
        while(n<nums.length){
            nums[n] = 0;
            n += 1;
        }
    }
}

File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void swap(vector<int> & arr, int i, int j){
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }
    void moveZeroes(vector<int>& nums) {
        int idx = 0;
        for(int i = 0; i < nums.size(); i++){
            if(nums[i] != 0) {
                swap(nums,idx,i);
                idx++;
            }
        }
        
        
    }
};